### ISSUE TITLE
Is it possible to build a CRM on this system?

### ISSUE THREAD

**Issue by pomazanbohdan** [webyaml/core#2]
Is it possible to build a CRM on this system?

**Comment by pomazanbohdan** [comment]
Add demo screenshot please

**Comment by Pomazan-Bogdan** [comment]
`system: users
 name:
  ru: Пользователи
  en: Users
 url: /users

field:

 id: 
  type: id
  
  
 data_create:
  type: date
  format: unixtime

 nickname:
  name:
   ru: Никнейм
   en: Nickname
  type: text
  long: 256
  
 login: 
  name:
    ru: Логин
    en: Login
  type: text
  long: 256

 password: 
  name:
    ru: Email
    en: Email
  type: hash
  
 email: 
  name:
   ru: Email
   en: Email
  type: text
  long: 256
  
 role: 
  name:
   ru: Роль
   en: Role
  type: ID
  table: roles

options:

 filtrable:[nickname, login, email]
 sortable:[nickname, login, email]
 `

**Comment by Pomazan-Bogdan** [comment]
[user.yaml.txt](https://github.com/webyaml/core/files/1981516/user.yaml.txt)

**Comment by webyaml** [comment]
Yes, it is possible to create a CRM using webyaml.  To do this you would need the following in your setup:
1) a database to save content and page urls.
2) a backend application to create pages and page content
3) a frontend page that will search the database for the requested page and content.  If content is found the page would show the content, otherwise it would show an error.

For content generation you could use any number of WYSIWYG html editors.  For the database, currently only mysql and sqlite are supported.  You could also create custom processors or use API calls to access other databases.

Many of the concepts used in webyaml are borrowed from CMS systems, in particular [TYPO3](https://typo3.org/).  However, when I created this system my concern was to be able to quickly make backend web applications for clients.  The applications had specific needs that were not compatible with the rigidness of a content management system. 

If you want to attempt to build a simple CMS using webyaml, that would be super.  Attached is a file that should help you on the task.  
[webyaml_as_cms.txt](https://github.com/webyaml/core/files/2199282/webyaml_as_cms.txt)

**Comment by webyaml** [comment]
Damn, i feel stupid.  I misread your question.  My response was for a CMS not a CRM.

I have not built a full blown CRM with webyaml, but i have used API calls to a SugarCRM/SweetCRM backend to make a custom interface.  

Still the answer is yes.  Webyaml is a like a high level programming language.  You can use it to build any type of web application.

Take a look at the rest docs: [http://webyaml.com/docs/reference/processors/rest](http://webyaml.com/docs/reference/processors/rest).  That will show you how to connect to a CRM via API.  After the data is retrieved you can display it on a page or use to create forms.  Send changes back to the CRM via API calls.
